Source/Redemption/Private/AudioManager.cpp

```diff
@@ -0,0 +1,83 @@
+#include "AudioManager.h"
+#include "Kismet/GameplayStatics.h"
+#include "Sound/SoundCue.h"
+#include "Components/AudioComponent.h"
+
+
+// Constants for random sound intervals, this is in seconds
+const float MinRandomSoundInterval = 3.0f; 
+const float MaxRandomSoundInterval = 15.0f; 
+
+// Sets default values
+AAudioManager::AAudioManager()
+{
+    // Set this actor to not call Tick() every frame. You can turn this on if you need it.
+    PrimaryActorTick.bCanEverTick = false;
+
+    // Initialize the MusicComponent and set it to not auto-activate
+    MusicComponent = CreateDefaultSubobject<UAudioComponent>(TEXT("MusicComponent"));
+    MusicComponent->bAutoActivate = false;
+
+    // Initialize the LoopingAmbientSoundComponent and set it to not auto-activate
+    LoopingAmbientSoundComponent = CreateDefaultSubobject<UAudioComponent>(TEXT("LoopingAmbientSoundComponent"));
+    LoopingAmbientSoundComponent->bAutoActivate = false;
+}
+
+// Called when the game starts or when spawned
+void AAudioManager::BeginPlay()
+{
+    Super::BeginPlay();
+
+    // Start playing the ambient music immediately if it's set
+    PlayAmbientMusic();
+
+    // Start playing the looping ambient sound
+    if (LoopingAmbientSoundComponent->Sound)
+    {
+        LoopingAmbientSoundComponent->Play(0.f);
+    }
+
+    // Start playing random ambient sounds
+    StartRandomAmbientSounds();
+}
+
+void AAudioManager::PlayRandomAmbientSound()
+{
+    int32 RandomIndex = FMath::RandRange(0, RandomAmbientSoundCues.Num() - 1);
+    UGameplayStatics::PlaySoundAtLocation(this, RandomAmbientSoundCues[RandomIndex], GetActorLocation());
+
+    // Reset the timer with a new random interval
+    float RandomInterval = FMath::RandRange(MinRandomSoundInterval, MaxRandomSoundInterval);
+    GetWorld()->GetTimerManager().SetTimer(RandomSoundTimerHandle, this, &AAudioManager::PlayRandomAmbientSound, RandomInterval, false);
+}
+
+void AAudioManager::StartRandomAmbientSounds()
+{
+
```

Comment:
so this function is called once to make sure that the RandomAmbientSoundCues is not empty and them after that PlayrandomAmbientSound will be called with random timers?

Reply:
Yea, the point of it is to set up the system for cues and checks if the RandomAmbientSoundCues array is not empty and gets PlayRandomAmbientSound to play sounds at random times.

Reply:
ok, seems good for me

Reply:
I realize what I said is basically what you said but in the form of an answer but, yes that is what it does haha